### ISSUE TITLE
ImportError: dynamic module does not define module export function (PyInit_PyKDL)

### ISSUE THREAD

**Issue by BCaran** [heuristicus/spot_ros#99]
Hi,
I am using ROS Melodic on Ubuntu 18.04 for NVIDIA Jetson Xavier AGX.

when I start:

`roslaunch spot_driver driver.launch`

i get import  import PyKDL error.

```
Traceback (most recent call last):
  File "/home/spot/catkin_ws/src/spot_ros/spot_driver/scripts/spot_ros", line 3, in <module>
    from spot_driver.spot_ros import SpotROS
  File "/home/spot/catkin_ws/src/spot_ros/spot_driver/src/spot_driver/spot_ros.py", line 23, in <module>
    import tf2_geometry_msgs
  File "/home/spot/catkin_ws/devel/lib/python3/dist-packages/tf2_geometry_msgs/__init__.py", line 34, in <module>
    exec(__fh.read())
  File "<string>", line 1, in <module>
  File "/home/spot/catkin_ws/src/geometry2/tf2_geometry_msgs/src/tf2_geometry_msgs/tf2_geometry_msgs.py", line 31, in <module>
    import PyKDL
ImportError: dynamic module does not define module export function (PyInit_PyKDL)
```

**Comment by heuristicus** [comment]
You're probably missing the `python3-pykdl` package, or `ros-noetic-tf2-kdl`. You can install it with apt.

For reference, here are some kdl packages I have on a machine which does not have this module missing.

```
dpkg -l | ag kdl
ii  liborocos-kdl-dev:amd64                                     1.4.0-7ubuntu1                                amd64        Kinematics and Dynamics Library development files
ii  liborocos-kdl1.4                                            1.4.0-7ubuntu1                                amd64        Kinematics and Dynamics Library runtime
ii  python3-pykdl:amd64                                         1.4.0-7ubuntu1                                amd64        Kinematics and Dynamics Library Python library
ii  ros-noetic-eigen-conversions                                1.13.2-1focal.20220926.184700                 amd64        Conversion functions between: - Eigen and KDL - Eigen and geometry_msgs.
ii  ros-noetic-kdl-conversions                                  1.13.2-1focal.20220926.190946                 amd64        Conversion functions between KDL and geometry_msgs types.
ii  ros-noetic-kdl-parser                                       1.14.2-1focal.20220926.195001                 amd64        The Kinematics and Dynamics Library (KDL) defines a tree structure to represent the kinematic and dynamic parameters of a robot mechanism.
ii  ros-noetic-tf2-kdl                                          0.7.5-1focal.20220926.225132                  amd64        KDL binding for tf2
```

**Comment by BCaran** [comment]
When I try to install it with sudo apt, I got error that there is no python3-pykdl.

**Comment by BCaran** [comment]
I fixed this issuse with installing Ubuntu 20.04 for Jetson Xavier AGX and installing ROS Noetic.

**Comment by heuristicus** [comment]
If you were not using noetic before, that may have been another part of the problem - this package does not support melodic or older versions of ROS.

It may be possible to use melodic with the instructions at https://github.com/heuristicus/spot_ros#building-for-melodic, but prefer use of noetic.

**Comment by BCaran** [comment]
I did some research and problem is that PyKDL is now  merged together for Python2 and Python3.

Usage of Noetic works like charm because Python 3 is official interpreter.

**Comment by danchoy** [comment]
To resolve this problem, I utilized Ubuntu 18.04 with Python versions 3.69 and 2.7. 
It was necessary to manually install all required modules. 
Additionally, the current version of orocos_kdl on Github was not compatible with my environment, 
so I opted to clone the previous version.

cd ~/catkin_ws/src
sudo apt update
sudo apt install libeigen3-dev libcppunit-dev
sudo apt install python3-psutil python3-future
python3 -m pip install psutil
python3 -m pip install future
git clone -b release-1.5 https://github.com/orocos/orocos_kinematics_dynamics.git
cd orocos_kinematics_dynamics/python_orocos_kdl
git submodule update --init
catkin build orocos_kdl
catkin build python_orocos_kdl
